core/tests/exhaustive_sync/trial.rs

```diff
@@ -110,117 +117,145 @@ impl Trial {
+                    let db = &self.devices_by_user[user_index][device_index];
                     let parent = find_by_name(db, &parent).id;
                     if let Err(err) = db.create_file(&name, parent, Document) {
                         self.status = Failed(format!("{:#?}", err));
                         break 'steps;
                     }
                 }
-                NewFolder { client, parent, name } => {
-                    let db = &self.clients[client];
+                NewFolder { user_index, device_index, parent, name } => {
+                    let db = &self.devices_by_user[user_index][device_index];
                     let parent = find_by_name(db, &parent).id;
                     if let Err(err) = db.create_file(&name, parent, Folder) {
                         self.status = Failed(format!("{:#?}", err));
                         break 'steps;
                     }
                 }
-                UpdateDocument { client, name, new_content } => {
-                    let db = &self.clients[client];
+                UpdateDocument { user_index, device_index, name, new_content } => {
+                    let db = &self.devices_by_user[user_index][device_index];
                     let doc = find_by_name(db, &name).id;
                     if let Err(err) = db.write_document(doc, new_content.as_bytes()) {
                         self.status = Failed(format!("{:#?}", err));
                         break 'steps;
                     }
                 }
-                RenameFile { client, name, new_name } => {
-                    let db = &self.clients[client];
+                RenameFile { user_index, device_index, name, new_name } => {
+                    let db = &self.devices_by_user[user_index][device_index];
                     let doc = find_by_name(db, &name).id;
                     if let Err(err) = db.rename_file(doc, &new_name) {
                         self.status = Failed(format!("{:#?}", err));
                         break 'steps;
                     }
                 }
-                MoveDocument { client, doc_name, destination_name } => {
-                    let db = &self.clients[client];
-                    let doc = find_by_name(db, &doc_name).id;
+                MoveFile { user_index, device_index, doc_name: non_folder_name, destination_name } => {
+                    let db = &self.devices_by_user[user_index][device_index];
+                    let non_folder = find_by_name(db, &non_folder_name).id;
                     let dest = find_by_name(db, &destination_name).id;
 
-                    if let Err(err) = db.move_file(doc, dest) {
+                    let move_file_result = db.move_file(non_folder, dest);
+                    match move_file_result {
+                        Ok(()) | Err(UiError(MoveFileError::LinkInSharedFolder)) | Err(UiError(MoveFileError::FolderMovedIntoItself)) => {}
```

Comment:
Because core resolves links, clients never know if what they're moving is a link, and therefore fuzzer won't know whether `LinkInSharedFolder` is acceptable. I just went ahead and combined `MoveDocument` and `MoveFolder` into this op assuming `FolderMovedIntoSelf` won't come up when moving a document (and `LinkInSharedFolder` won't come up when moving a non-link)